context/PokemonState.tsx

```diff
@@ -2,32 +2,49 @@ import React, { useReducer } from 'react';
 import PokemonContext from './pokemonContext';
 import PokemonReducer from './pokemonReducer';
 import { Pokemon } from '../api/pokemon';
-
-export interface PokemonState {
-  favPokemon: Pokemon | null;
-}
-
-export enum PokemonActionType {
-  SAVE_FAV_POKEMON = 'SAVE_FAV_POKEMON'
-}
-export type PokemonAction = { type: PokemonActionType; payload: any };
+import {
+  PokemonActionTypes,
+  PokemonStateInterface
+} from './PokemonContextUtils';
+import { AsyncStorage } from 'react-native';
 
 const PokemonState = (props: any) => {
-  const initialState: PokemonState = {
+  const initialState: PokemonStateInterface = {
     favPokemon: null
   };
 
   const [state, dispatch] = useReducer(PokemonReducer, initialState);
 
+  const POKEMON_STORAGE_KEY = 'FAV_POKEMON';
+
+  // Load Favourite Pokemon
+  async function loadFavPokemon() {
+    try {
+      const fpokemon = await AsyncStorage.getItem(POKEMON_STORAGE_KEY);
```

Comment:
nit -> I'd use `pokemon` or `favPokemon` in both save/load functions for consistency